x/config/config.go

```diff
@@ -171,3 +174,79 @@ func NewPacketListener(transportConfig string) (transport.PacketListener, error)
 		return nil, fmt.Errorf("config scheme '%v' is not supported", url.Scheme)
 	}
 }
+
+func SanitizeConfig(transportConfig string) (string, error) {
+	// Do nothing if the config is empty
+	if transportConfig == "" {
+		return "", nil
+	}
+	// Split the string into parts
+	parts := strings.Split(transportConfig, "|")
+
+	// Iterate through each part
+	for i, part := range parts {
+		u, err := parseConfigPart(part)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse config part: %w", err)
+		}
+		scheme := strings.ToLower(u.Scheme)
+		switch scheme {
+		case "ss", "socks5", "vless":
+			parts[i], _ = sanitizeURL(u)
+		case "split", "tls", "tlsfrag":
```

Comment:
@fortuna do you mean define a specific sanitizer for `socks5`?

Reply:
Since `socks5` URL has a known format like this `socks5://myuser:mypassword@192.168.1.100:1080`, we can just look for `UserInfo` and redact it. If it does not exist, we just return the URL as is.


```go
func sanitizeSocks5URL(u *url.URL) (string, error) {
	const redactedPlaceholder = "REDACTED"
	if u.User != nil {
		u.User = url.User(redactedPlaceholder)
		return u.String(), nil
	}
	return u.String(), nil
}
```

Current `sanitizeURLGeneric` removes IP and port info from `socks5`.